hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/RatisUnderReplicationHandler.java

```diff
@@ -217,7 +217,15 @@ private List<DatanodeDetails> getSources(
           }
         })
         .filter(r -> !pendingDeletion.contains(r.getDatanodeDetails()))
-        .sorted((r1, r2) -> r2.getSequenceId().compareTo(r1.getSequenceId()))
+        .collect(Collectors.toList());
+
+    // We should replicate only the max available sequence ID, as replicas with
+    // earlier sequence IDs may be stale copies.
+    long maxSequenceId = availableSources.stream()
+        .map(ContainerReplica::getSequenceId).max(Long::compareTo).orElse(0L);
```

Comment:
`Long getSequenceId()` may return `null`, so for safety we may add a filter in the stream collecting `availableSources` above.

```
        .filter(r -> r.getSequenceId() != null)
```

Not strictly necessary, as I'm not sure all other code handles `null` gracefully.

Reply:
Probably is worth adding this "just incase". It might even be better to change the map step to return zero if getSequenceId returns null. That way, if for some strange reason all replicas don't have a sequenceID we just use any of them, rather than filter all of them out.

Reply:
> might even be better to change the map step to return zero if getSequenceId returns null. That way, if for some strange reason all replicas don't have a sequenceID we just use any of them, rather than filter all of them out.

Good point.